pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -214,18 +214,10 @@ public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceE
         // decrement unack-message count for removed consumer
         addUnAckedMessages(-consumer.getUnackedMessages());
         if (consumerSet.removeAll(consumer) == 1) {
```

Comment:
Is there a need to continue to have conditional logic based on this?

Reply:
> Is there a need to continue to have conditional logic based on this?

Good question, I think this conditional logic is still needed.

There are other operations after removing a consumer success:
- Move the pending acked messages into `redeliveryMessages`
- Update `totalAvailablePermits`
- Trigger a `readMoreEntries`

For the scenario that `consumerList` has two same consumer, and `consumerSet` only has one, the method `removeConsumer` will be called by `dispatcher.disconnectAllConsumers` twice, but the broker should only trigger the operations above once.